### PR TITLE
Add view

### PR DESCRIPTION
**Pull request by wolterbwambale**
### Project requirements
- Implement the design from the sneak peek wireframes.
    - Styling is nice to have but focus on making sure everything functions first. (Note: Ignore login logic, you would be using Devise in coming milestone)

**Note: Analyze the wireframes deeply and try to identify all the necessary elements and how a user would navigate from one page to another**
    
All Users page wireframe Single User page wireframe All Posts by a given User page wireframe Single Post page wireframe

- Use methods that you have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).
- **Hint:** In order to see some posts or comments, you might need to add some data to your database as described [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md#use-models-to-insert-data).

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -1 +1,13 @@
-<h1>Here is a list of posts for a given user</h1>
+<%= render 'users/user' %>
+<% @posts.each do |post| %>
+  <a href="<%= user_post_path(post.author, post) %>"><%= render post %></a>
+  <%= "no comments for this post" if post.comments.empty? %>
+    <div class="comments">
+    <% post.recent_comments.each do |comment|%>
```
Comment (joel-kalema) [inline app/views/posts/index.html.erb:6]: - [ ] Good job working on views, however, make sure you are displaying correctly the `author's name`. Kindly fix it 👍 

> | | template | | Your project | |
> |-|---------|-|--------|-|
> | |blog_user_all_posts| |Screenshot 2023-11-02 014654| |

File: app/views/posts/show.html.erb [hunk 2]
```diff
@@ -1 +1,7 @@
-<h1>Here is a list of posts for a given user</h1>
+<%= render partial: 'post', locals: { post: @post } %> <br>
+<%= "no comments for this post" if @post.comments.empty?%>
+<div class="comments">
+<% @post.comments.each do |comment|%>
+        <%=render comment %>
```
Comment (joel-kalema) [inline app/views/posts/show.html.erb:5]: - [ ] Also there, Kindly fix it 👍 
> | | template | | Your project | |
> |-|---------|-|--------|-|
> | |blog_single_post| |Screenshot 2023-11-02 014744| |

**Review by joel-kalema — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @wolterbwambale 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

0bd223a2b156e0eaf61a713e84c3d3b7



## To highlight :champagne: 

- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me @joel-kalema  when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @wolterbwambale, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**